pycheribuild/projects/cross/cheribsd.py

```diff
@@ -1678,6 +1678,8 @@ def default_kernel_config(self, platform: "Optional[ConfigPlatform]" = None, **f
 
     def extra_kernel_configs(self) -> "list[CheriBSDConfig]":
         # Everything that is not the default kernconf
+        if self.kernel_config != self.default_kernel_config():
```

Comment:
What if I do --cheribsd/kernel-config=DEFAULT_KERNEL_CONFIG_HERE? That should override it too, surely?

Reply:
Yes and it should, this should only affect the extra kernels that are added on top of the default kernel. I think that the `kernel-config` argument is already taken into account from the `default_kernel` side.

Reply:
Today, `--cheribsd-morello-purecap/kernel-config=GENERIC-MORELLO cheribsd-morello-purecap` builds GENERIC-MORELLO and GENERIC-MORELLO-PURECAP, whilst `--cheribsd-morello-purecap/kernel-config=GENERIC-MORELLO-PURECAP cheribsd-morello-purecap` builds GENERIC-MORELLO-PURECAP twice

Reply:
Yeah I don't think that we should build the `GENERIC-MORELLO-PURECAP` in this case, if we specify a kernel-config to override. It feels like it is ignoring the override, and also building the kernel twice is wrong.